File: backend/src/main/java/com/prescription/backend/Model/Prescription.java
package com.prescription.backend.Model;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor

@Entity
@Table(name = "prescriptions")
public class Prescription {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "prescription_id")
    private Long prescriptionID; // Surrogate primary key


    @Column(name = "prescription_name")
    private String prescriptionName;

    @ManyToOne
    @JoinColumn(name = "patient_id", nullable = false)
    private Patient patient;

    @ManyToOne
    @JoinColumn(name = "doctor_id", nullable = false)
    private Doctor doctor;

    @ManyToOne
    @JoinColumn(name = "medication_id", nullable = false)
    private Medication medication;

    @Column(name = "prescription_date")
    private String prescriptionDate;

    @Column(name = "quantity")
    private int quantity;

    @Column(name = "instructions", length = 100)
    private String instructions;
}

File: prescription-app/src/app/model/Prescription.ts
// prescription.model.ts
import { Doctor } from "./Doctor";
import { Medication } from "./Medication";
import { Patient } from "./Patient";

export interface Prescription {
  prescriptionID: number;
  prescriptionName: string;
  patient: Patient;
  doctor: Doctor;
  prescriptionDate: string;
  quantity: number;
  instructions: string;
  medication: Medication;
}
